R/approxfun-methods.R

```diff
@@ -6,7 +6,20 @@ setMethod(f="approxfun",
   if (isEmpty(x)) {
     function(x)rep.int(NA, length(x))
   } else {
-    approxfun(x=x@mass, y=x@intensity, method=method,
+    approxfun(x=mass(x), y=intensity(x), method=method,
               yleft=yleft, yright=yright, rule=rule, f=f, ties=ties)
   }
 })
+
+## MassSpectrumOnDisk
+setMethod(f="approxfun",
+          signature=signature(x="MassSpectrumOnDisk"),
```

Comment:
You mean that instead of defining a method for every `MassSpectrumOnDisk`, simply use option #1 and define all methods for `AbstractMassSpectrumObject`?